src/monattack.cpp

```diff
@@ -5283,6 +5284,36 @@ bool mattack::stretch_attack( monster *z )
     return true;
 }
 
+bool mattack::zombie_fuse( monster *z )
+{
+    monster *critter = nullptr;
+    for( const tripoint &p : g->m.points_in_radius( z->pos(), 1 ) ) {
+        critter = g->critter_at<monster>( p );
+        if( critter != nullptr &&  critter->faction == z->faction
+            && critter != z && critter->get_size() <= z->get_size() ) {
+            break;
+        }
+    }
+
+    if( critter == nullptr ||
+        ( z->get_hp() + critter->get_hp() > z->get_hp_max() +
+          effect_grown_of_fuse.obj().get_max_intensity() ) ) {
+        return false;
+    }
+    if( g->u.sees( *z ) ) {
+        g->u.add_msg_if_player( _( "The %1$s fuses with the %2$s." ),
```

Comment:
```suggestion
        add_msg( _( "The %1$s fuses with the %2$s." ),
```
Calling `add_msg_if_player` *on the player object* is very redundant. This if effectively like
```C++
if( g->u.is_player() ) {
    add_msg( ... );
```
And of course, `g->u` is the player character, so that condition is always true.